File: runtime/perl/units/wasi/fd_sync.pl
# requires:
use IO::Handle ();

sub wasi_fd_sync {
    my ($self, $fd) = @_;
    my $e = $self->{fds}{$fd};
    return ERRNO_BADF if !defined($e) || $e->{dir};
    return ERRNO_IO unless defined $e->{fh}->sync;
    return ERRNO_SUCCESS;
}

File: crates/dewasm-backend-python/units/wasi/fd_sync.py
def wasi_fd_sync(self, fd):
    io = self.fds.get(fd)
    if io is None or isinstance(io, self.WasiDir):
        return self.ERRNO_BADF
    try:
        os.fsync(io.fileno())
    except OSError:
        return self.ERRNO_IO
    return self.ERRNO_SUCCESS
